### PR TITLE
svthevc: add SVT-HEVC to ffmpeg

### PR DESCRIPTION
**Pull request by 1480c1**
Needs to be tested as the patches are based off the 4.1 branch

### CODE REVIEW

**Comment by 1480c1** [comment]
Although I do see the patches at patchwork from jsunintel, I'm not sure if it would be a good idea to use those mainly because those seem to be updated a lot

**Comment by wiiaboo** [comment]
I think it's better to check if ffmpeg_options.txt contains `--enable-libsvthevc` instead than always adding it with `svthevc=yes` (and apply the patch only if the option is there, or remove it if the patch fails to apply to prevent configure breakage). People might want the standalone encoder on its own or included in x265 but not include it in FFmpeg.

**Comment by 1480c1** [comment]
Okay, should I leave a note on the bat file to add the flag if they want to use svt-hevc with ffmpeg?

**Comment by wiiaboo** [comment]
Sure.

**Comment by wiiaboo** [comment]
There's also the ffmpeg_options_64.txt hack, but that's replacing the common one, not incrementing it.

File: build/media-suite_compile.sh [hunk 1]
```diff
@@ -1224,7 +1224,7 @@ fi
 
 _check=(SvtHevcEnc.pc libSvtHevcEnc.dll.a svt-hevc/EbApi.h
     bin-video/{libSvtHevcEnc.dll,SvtHevcEncApp.exe})
-if [[ $svthevc = "y" ]] && do_vcs "https://github.com/OpenVisualCloud/SVT-HEVC.git"; then
+if [[ $bits != "32bits" && $svthevc = "y" ]] && do_vcs "https://github.com/OpenVisualCloud/SVT-HEVC.git"; then
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: It's `32bit` actually. But might as well use `$bits = "64bit"`.

Comment (wiiaboo) [reply build/media-suite_compile.sh]:
```suggestion
if [[ $bits = "32bit" ]]; then
    do_removeOption --enable-libsvthevc
elif { [[ $svthevc = "y" ]] || enabled libsvthevc; } &&
    do_vcs "https://github.com/OpenVisualCloud/SVT-HEVC.git"; then
```

File: build/media-suite_compile.sh [hunk 2]
```diff
@@ -1709,6 +1709,13 @@ if [[ $ffmpeg != "no" ]]; then
             do_patch "https://0x0.st/zeB6.txt"
         fi
 
+        if enabled libsvthevc && [[ $bits != "32bits"]]; then
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: Then you can leave just the check for `enabled libsvthevc` here.

Comment (1480c1) [reply build/media-suite_compile.sh]: Done

**Pull request merged**